editor/filesystem_dock.cpp

```diff
@@ -909,10 +909,21 @@ void FileSystemDock::_file_option(int p_option) {
 			OS::get_singleton()->shell_open(String("file://") + dir);
 		} break;
 		case FILE_OPEN: {
-			int idx = files->get_current();
-			if (idx < 0 || idx >= files->get_item_count())
-				break;
-			_select_file(idx);
+			if (files->get_selected_items().size() > 1) {
+				for (int i = 0; i < files->get_item_count(); i++) {
```

Comment:
Can't you iterate directly over `files->get_selected_items()` and avoid the `is_selected` check?

Reply:
In fact, I think it can use `_select_file(i)` in `for` loop directly. 
Avoid the `files->get_selected_items().size() > 1` check.
But still use `is_selected` check.
So even if it is select one file. It's still work fine too.

Reply:
> In fact, I think it can use _select_file(i) in for loop directly.

Yeah I thought about this too, I just wondered what would happen if you select several folders, or a mix of scenes and resources and folders. If it works fine, using `_select_file(i)` would definitely be better than reimplementing part of its functionality.

> Avoid the files->get_selected_items().size() > 1 check.
But still use is_selected check.
So even if it is select one file. It's still work fine too.

Ah I get it, you need to fetch the right `i` to get the metadata, so you have to iterate over the whole `files->get_item_count()`.

Reply:
Yep, it can't get select files index directly.I must do it myself.
I'm testing and try to find a better way at this night.
But I have to eat dinner first...
Now the sun has fallen.
![b](https://user-images.githubusercontent.com/24988459/33545829-4e99f16c-d91a-11e7-9880-1411a4011f50.png)